tests/integration/test_persistence.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from datetime import datetime, timedelta, timezone
+from pathlib import Path
+
+import pytest
+import yaml
+from helpers import get_unit_address, is_alertmanager_up
+from pytest_operator.plugin import OpsTest
+
+from alertmanager_client import Alertmanager
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+app_name = METADATA["name"]
+resources = {"alertmanager-image": METADATA["resources"]["alertmanager-image"]["upstream-source"]}
+
+
+@pytest.mark.abort_on_fail
+async def test_silences_persist_across_upgrades(ops_test: OpsTest, charm_under_test, httpserver):
+    # deploy alertmanager charm from charmhub
+    logger.info("deploy charm from charmhub")
+    await ops_test.model.deploy("ch:alertmanager-k8s", application_name=app_name, channel="edge")
```

Comment:
Iiuc, for this test we can deploy the charm-under-test and then refresh with the same again, i.e. no need for a charmhub charm.

Reply:
Is there any strong reason to prefer doing it this way. I wanted to make the test as realistic as possible, in that the two versions of the charm being deployed are actually different. Besides by deploying from charmhub and upgrading from git we ensure detection if any future PR breaks persistence due to any incompatibility of currently released charm and proposed PR.